### PR TITLE
#161051781 User can delete a request with 'open' status

### PR DESCRIPTION
**Pull request by matthewacha**
#### What does this PR do?
This PR enables a requester to delete a request, their manager is then notified about the action carried out.

#### Description of Task to be completed?
-Travel requester should be able to delete only open requests
- The request should be deleted from Manager's approval page
- The manager should get an in-app notification to this effect

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Enter into the project directory - `cd travel_tool_back`
- checkout the branch `ft-delete-requests-161051781`
- Start the server - `make start`
- Navigate to [travela](http://travela-local.andela.com:3000/#) in your browser and log in
- Login on your browser and grab your token from the application tab in Developers tab
- Open Postman
- Add your token in headers
- Navigate to your database GUI (pgAdmin or postico)
- Update your user role in UserRoles table to 'Travel Administrator' `29187`
- Post a request as shown [here](https://github.com/andela/travel_tool_back/pull/44)
- Delete the request using `http://127.0.0.1:5000/api/v1/requests/requestId`

#### What are the relevant pivotal tracker stories?
[161051781](https://www.pivotaltracker.com/story/show/161051781)

#### Screenshots (if appropriate)
image

### CODE REVIEW

File: src/modules/requests/RequestsController.js [hunk 1]
```diff
@@ -341,6 +349,59 @@ class RequestsController {
       return Error.handleError(error, 500, res);
     }
   }
+
+  static async handleDestroyTripComments(req, res) {
+    const { requestId } = req.params;
+    try {
+      const trips = await TripsController.getTripsByRequestId(requestId, res);
```
Comment (amarachukwu-agbo) [inline src/modules/requests/RequestsController.js]: @matthewacha , finding trips associated with a request this way before mapping and calling the destroy method on each trip might not be the most efficient way to go about this. The same result can be achieved with one query like so:
eg 

models.Trip.destroy({
where: {
requestId
}
});

File: src/modules/requests/RequestsController.js [hunk 2]
```diff
@@ -341,6 +349,59 @@ class RequestsController {
...
+      await trips.forEach((trip) => { trip.destroy(); });
+      const comments = await models.Comment.findAll(
+        { where: { requestId } }
+      );
+      if (comments) {
+        await comments.forEach((comment) => { comment.destroy(); });
```
Comment (amarachukwu-agbo) [inline src/modules/requests/RequestsController.js]: Please apply the same feedback given above here

**Comment by amarachukwu-agbo** [comment]
@matthewacha, I noticed that the approval record for a request is not deleted when a request is deleted. Please replicate what you did for the Trip and Comment models for the Approval model.

**Comment by amarachukwu-agbo** [comment]
In the description of task to be completed, you have included `There should be a pop up modal asking the user if they are sure they want to delete the travel request`. Is this not for the front-end though?

File: src/modules/requests/RequestsController.js (hunk 1 above)
Comment (matthewacha) [reply src/modules/requests/RequestsController.js]: Done @amarachukwu-agbo

File: src/modules/requests/RequestsController.js (hunk 2 above)
Comment (matthewacha) [reply src/modules/requests/RequestsController.js]: Done @amarachukwu-agbo

**Comment by amarachukwu-agbo** [comment]
@matthewacha , please fix failing tests

**Pull request merged**